### PR TITLE
[FLINK-7402] Fix ineffective null check in NettyMessage#write()

### PR DESCRIPTION
**Pull request by zjureel**
## What is the purpose of the change

Fix ineffective null check in NettyMessage#write()

## Brief change log

*(for example:)*
  - *Add null check and remove unnecessary null check*

## Verifying this change

*(Please pick either of the following options)*

No test case

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java [hunk 1]
```diff
@@ -220,6 +220,10 @@ void releaseBuffer() {
 
 		@Override
 		ByteBuf write(ByteBufAllocator allocator) throws IOException {
+			if (null == buffer) {
+				throw new NullPointerException();
```
Comment (tedyu) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java]: Add exception message to provide more information.

Comment (NicoK) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java]: actually, better use `Preconditions#checkNotNull(T)` - an additional error message there usually does not make too much sense as the failure is clear

**Comment by zjureel** [comment]
@tedyu @NicoK Than you for your suggestions, it sounds good to me, thanks

File: flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java [hunk 2]
```diff
@@ -220,9 +220,7 @@ void releaseBuffer() {
 
 		@Override
 		ByteBuf write(ByteBufAllocator allocator) throws IOException {
-			if (null == buffer) {
-				throw new NullPointerException();
-			}
+			Preconditions.checkNotNull(buffer, "When deserializing the buffer should have to be requested first.");
```
Comment (NicoK) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java]: Actually, the message should rather be `No buffer instance to serialize.` since the constructor leading to this code path is only used for de-serialization.

Comment (zjureel) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java]: Great, thank you for your suggestion :)

File: flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java [hunk 3]
```diff
@@ -18,6 +18,12 @@
 
 package org.apache.flink.runtime.io.network.netty;
 
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.ByteBuffer;
+import java.util.List;
+
```
Comment (NicoK) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java]: sorry for bothering you again, but please revert this change in the import order (it's also against the order defined by our checkstyle)

**Comment by zjureel** [comment]
@NicoK Than you for your review, I have reverted the imports changed by IDE, thanks

**Comment by zjureel** [comment]
@NicoK Great, tt's too useful for me, thank you very much!

**Comment by tedyu** [comment]
lgtm

**Comment by greghogan** [comment]
Merging ...
